### PR TITLE
create Input Types And Form Elements

### PR DESCRIPTION
**Pull request by danielrileyde**
# Context

This PR aims to tackle [More input types & Forms - A11y](https://github.com/spiced-academy/honey-web-dev/blob/main/sessions/html-forms/challenges-html-forms.md).

# Changes

- [ ] updated A11y labels
- [ ] created email input field and label
- [ ] created date input field and label
- [ ] created 3 radio buttons and subsequent labels
- [ ] created concerns text area and label
- [ ] created 2 check boxes for news letter and data consent with labels
- [ ] created 2 buttons one for reset data and one to submit data

# How to run 

1. Clone repo
2. Checkout this branch
3. Run page in browser

### CODE REVIEW

File: html/html-forms/html-forms_Finding-a11y-errors/index.html
Comment (reverbePrintemps) [inline html/html-forms/html-forms_Finding-a11y-errors/index.html:1]: Did you mean to lump this file along with the other in this PR?

File: html/html-forms/html-forms_zoo/index.html [hunk 1]
```diff
@@ -21,20 +21,78 @@ <h2 id="form-heading">Zoo Visitor Sign Up</h2>
...
-          <!-- Date: label & input -->
+          <form>
+            <label for="email-address">Email Address</label>
+            <input
+              type="email"
+              placeholder="yourname@example.com"
+              name="email-address"
+              id="email-address"
+            />
+            <!-- Date: label & input -->
```
Comment (reverbePrintemps) [inline html/html-forms/html-forms_zoo/index.html:32]: I recommend not leaving "simple" (ie. for things that are rather explicit/self-explanatory) comments in a shared codebase. But your call : )

File: html/html-forms/html-forms_zoo/index.html [hunk 2]
```diff
@@ -21,20 +21,78 @@ <h2 id="form-heading">Zoo Visitor Sign Up</h2>
...
+            <label for="arrival-date">Date of Arrival</label>
+            <input type="date" name="arrival-date" id="arrival-date" />
+          </form>
 
           <fieldset class="radio-options">
             <legend>Which one of these animals would you like to visit?</legend>
 
             <!-- Radio Buttons: labels & inputs -->
+            <label for="relaxed sloths">Relaxed Slothes</label>
```
Comment (reverbePrintemps) [inline html/html-forms/html-forms_zoo/index.html:41]:
```suggestion
            <label for="relaxed-sloths">Relaxed Slothes</label>
```

File: html/html-forms/html-forms_zoo/index.html [hunk 3]
```diff
@@ -21,20 +21,78 @@ <h2 id="form-heading">Zoo Visitor Sign Up</h2>
...
+            <input
+              type="radio"
+              name="special-attraction"
+              id="relaxed sloths"
+              value="sloths"
+            />
+
+            <label for="jumpy kangaroos">Jumpy Kangaroos</label>
```
Comment (reverbePrintemps) [inline html/html-forms/html-forms_zoo/index.html:49]:
```suggestion
            <label for="jump- kangaroos">Jumpy Kangaroos</label>
```

File: html/html-forms/html-forms_zoo/index.html [hunk 4]
```diff
@@ -21,20 +21,78 @@ <h2 id="form-heading">Zoo Visitor Sign Up</h2>
...
+            <input
+              type="radio"
+              name="special-attraction"
+              id="jumpy kangaroos"
```
Comment (reverbePrintemps) [inline html/html-forms/html-forms_zoo/index.html:53]:
```suggestion
              id="jumpy-kangaroos"
```

File: html/html-forms/html-forms_zoo/index.html [hunk 5]
```diff
@@ -21,20 +21,78 @@ <h2 id="form-heading">Zoo Visitor Sign Up</h2>
...
+              value="kangaroos"
+            />
+
+            <label for="thieving racoons">Thieving Racoons</label>
```
Comment (reverbePrintemps) [inline html/html-forms/html-forms_zoo/index.html:57]:
```suggestion
            <label for="thieving-racoons">Thieving Racoons</label>
```

File: html/html-forms/html-forms_zoo/index.html [hunk 6]
```diff
@@ -21,20 +21,78 @@ <h2 id="form-heading">Zoo Visitor Sign Up</h2>
...
+            <input
+              type="radio"
+              name="special-attraction"
+              id="thieving racoons"
```
Comment (reverbePrintemps) [inline html/html-forms/html-forms_zoo/index.html:61]:
```suggestion
              id="thieving-racoons"
```

File: html/html-forms/html-forms_zoo/index.html [hunk 7]
```diff
@@ -21,20 +21,78 @@ <h2 id="form-heading">Zoo Visitor Sign Up</h2>
...
+          /></label>
+
+          <label for="sign-up-newsletter"
+            >Sign up to out newsletter
+            <input
+              type="checkbox"
+              name="sign-up-newsletter"
+              id="sign-up-newsletter"
+            />
+          </label>
```
Comment (reverbePrintemps) [inline html/html-forms/html-forms_zoo/index.html:91]: Formatting/indentation could be improved here imo, for example: 
```suggestion
          <label for="accept-terms-conditions">
            I accept the terms and conditions
            <input
              type="checkbox"
              name="accept-terms-conditions"
              id="accept-terms-conditions"
            />
          </label>
```

File: html/html-forms/html-forms_zoo/index.html [hunk 8]
```diff
@@ -21,20 +21,78 @@ <h2 id="form-heading">Zoo Visitor Sign Up</h2>
...
+
           <!-- Two Buttons: reset & submit -->
+          <button type="'reset">Reset Form</button>
```
Comment (reverbePrintemps) [inline html/html-forms/html-forms_zoo/index.html:94]:
```suggestion
          <button type="reset">Reset Form</button>
```

**Review by reverbePrintemps — CHANGES_REQUESTED** [review]
Mostly great but I'd request some changes around naming consistency (kebab case), a couple formatting nitpicks and the technically broken "reset" button. 👍

File: html/html-forms/html-forms_zoo/index.html (hunk 8 above)
Comment (danielrileyde) [reply html/html-forms/html-forms_zoo/index.html:94]: good spot!

**Comment by danielrileyde** [comment]
okay, thanks, ill sort out the kebab casing and naming conventions.

**Pull request merged**